nixos/modules/services/backup/tsm.nix

```diff
@@ -87,16 +87,35 @@ in
       environment.DSM_LOG = "/var/log/tsm-backup/";
       # TSM needs a HOME dir to store certificates.
       environment.HOME = "/var/lib/tsm-backup";
-      # for exit status description see
-      # https://www.ibm.com/support/knowledgecenter/en/SSEQVQ_8.1.8/client/c_sched_rtncode.html
-      serviceConfig.SuccessExitStatus = "4 8";
-      # The `-se` option must come after the command.
-      # The `-optfile` option suppresses a `dsm.opt`-not-found warning.
-      serviceConfig.ExecStart =
-        "${cfgPrg.wrappedPackage}/bin/dsmc ${cfg.command} -se='${cfg.servername}' -optfile=/dev/null";
-      serviceConfig.LogsDirectory = "tsm-backup";
-      serviceConfig.StateDirectory = "tsm-backup";
-      serviceConfig.StateDirectoryMode = "0750";
+      serviceConfig = {
+        # for exit status description see
+        # https://www.ibm.com/docs/en/spectrum-protect/8.1.13?topic=clients-client-return-codes
+        SuccessExitStatus = "4 8";
+        # The `-se` option must come after the command.
+        # The `-optfile` option suppresses a `dsm.opt`-not-found warning.
+        ExecStart =
+          "${cfgPrg.wrappedPackage}/bin/dsmc ${cfg.command} -se='${cfg.servername}' -optfile=/dev/null";
+        LogsDirectory = "tsm-backup";
+        StateDirectory = "tsm-backup";
+        StateDirectoryMode = "0750";
+        # systemd sandboxing
```

Comment:
Where did these values come from? Experimentation? Upstream documentation? How do we have confidence this won't break existing installs?

Reply:
This is derived from experimentation.  The documetation contains limited information about systemd (and the package even two service file, albeit not for the "normal" backup program), but there is no sandboxing or information about it.  You're right, there is no guarantee this won't break existing installations.

According to my experiments, a "regular" backup (covering the filesystem, up to API filesystems) works fine with these settings.  I don't expect other use-cases that are driven with this module, but I'm not sure.  On the other hand, these sandboxing settings likely only protect from bugs, not from malicious code, as the service still has to run with root privileges.  So I expect limited use, but also a low potential for harm by these changes.

If you consider these options too risky, I can drop some of them (or all) from the branch.

Reply:
I think it is a good idea to keep these options. I just wanted to have some clarification on where they came from. This can bake in unstable for the next ~5 months and hopefully if there are any issues we'll catch them before the next release. Looks good :+1: